samples/python2/calibrate.py

```diff
@@ -12,64 +28,88 @@
 # built-in modules
 import os
 
-
-USAGE = '''
-USAGE: calib.py [--save <filename>] [--debug <output path>] [--square_size] [<image mask>]
-'''
-
-
-
 if __name__ == '__main__':
     import sys
     import getopt
     from glob import glob
 
-    args, img_mask = getopt.getopt(sys.argv[1:], '', ['save=', 'debug=', 'square_size='])
+    args, img_mask = getopt.getopt(sys.argv[1:], '', ['debug=', 'square_size='])
     args = dict(args)
-    try:
+    args.setdefault('--debug', './output/')
+    args.setdefault('--square_size', 1.0)
+    if not img_mask:
+        img_mask = '../data/left*.jpg'  # default
+    else:
         img_mask = img_mask[0]
-    except:
-        img_mask = '../data/left*.jpg'
 
     img_names = glob(img_mask)
     debug_dir = args.get('--debug')
-    square_size = float(args.get('--square_size', 1.0))
+    if not os.path.isdir(debug_dir):
+        os.mkdir(debug_dir)
+    square_size = float(args.get('--square_size'))
 
     pattern_size = (9, 6)
-    pattern_points = np.zeros( (np.prod(pattern_size), 3), np.float32 )
-    pattern_points[:,:2] = np.indices(pattern_size).T.reshape(-1, 2)
+    pattern_points = np.zeros((np.prod(pattern_size), 3), np.float32)
+    pattern_points[:, :2] = np.indices(pattern_size).T.reshape(-1, 2)
     pattern_points *= square_size
 
     obj_points = []
     img_points = []
     h, w = 0, 0
+    img_names_undistort = []
     for fn in img_names:
-        print('processing %s...' % fn,)
+        print('processing %s... ' % fn, end='')
         img = cv2.imread(fn, 0)
         if img is None:
-          print("Failed to load", fn)
-          continue
+            print("Failed to load", fn)
+            continue
 
         h, w = img.shape[:2]
         found, corners = cv2.findChessboardCorners(img, pattern_size)
         if found:
-            term = ( cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_COUNT, 30, 0.1 )
+            term = (cv2.TERM_CRITERIA_EPS + cv2.TERM_CRITERIA_COUNT, 30, 0.1)
             cv2.cornerSubPix(img, corners, (5, 5), (-1, -1), term)
+
         if debug_dir:
             vis = cv2.cvtColor(img, cv2.COLOR_GRAY2BGR)
             cv2.drawChessboardCorners(vis, pattern_size, corners, found)
             path, name, ext = splitfn(fn)
-            cv2.imwrite('%s/%s_chess.bmp' % (debug_dir, name), vis)
+            outfile = debug_dir + name + '_chess.png'
+            cv2.imwrite(outfile, vis)
+            if found:
+                img_names_undistort.append(outfile)
+
         if not found:
             print('chessboard not found')
             continue
+
         img_points.append(corners.reshape(-1, 2))
         obj_points.append(pattern_points)
 
         print('ok')
 
+    # calculate camera distortion
     rms, camera_matrix, dist_coefs, rvecs, tvecs = cv2.calibrateCamera(obj_points, img_points, (w, h), None, None)
-    print("RMS:", rms)
-    print("camera matrix:\n", camera_matrix)
-    print("distortion coefficients: ", dist_coefs.ravel())
+
+    # print("RMS:", rms)
```

Comment:
- Instead of printing to console, the undistorted images are now calculated and saved to disk, so the found calibration is not the final result of the example anymore.
- Could have deleted the `print` lines as well, but wanted to keep the convenient possibility to print them again for people working with the script.

If you want the values to be printed I change that and include the `print` again.